notebooks/docker_local_import.ipynb

```diff
@@ -0,0 +1,426 @@
+    "    pdf_files = find_recursive(downloads, \"pdf\")\n",
+    "    html_files = find_recursive(downloads, \"html\")\n",
+    "    ret = []\n",
+    "    for i in pdf_files + html_files:\n",
+    "        t = reimport_timestamp(root, i[\"suffix\"])\n",
+    "        if t > 0:\n",
+    "            i[\"timestamp\"] = t\n",
+    "            ret.append(i)\n",
+    "        else:\n",
+    "            # print('File unchanged', i['suffix'])\n",
+    "            pass\n",
+    "\n",
+    "    return ret"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "id": "851a6064-ae13-4b59-8d48-07882edfc590",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/app/work\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100 2797k  100 2797k    0     0  2722k      0  0:00:01  0:00:01 --:--:-- 2723k\n",
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100 62509  100 62509    0     0   139k      0 --:--:-- --:--:-- --:--:--  139k\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "pwd\n",
+    "sudo mkdir -p /app/work/crawl_data/downloads/pdf\n",
+    "sudo mkdir -p /app/work/crawl_data/downloads/html\n",
+    "sudo curl https://sortbenchmark.org/ELSAR2022.pdf -o /app/work/crawl_data/downloads/pdf/elsar.pdf\n",
+    "sudo curl https://sortbenchmark.org -o /app/work/crawl_data/downloads/html/home.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "id": "84ec0d42-b97e-4209-9ea3-6cd967ef680a",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[{'path': PosixPath('/app/work/crawl_data/downloads/pdf/elsar.pdf'),\n",
+       "  'suffix': PosixPath('pdf/elsar.pdf'),\n",
+       "  'type': 'pdf',\n",
+       "  'timestamp': 1707347951.733037},\n",
+       " {'path': PosixPath('/app/work/crawl_data/downloads/html/home.html'),\n",
+       "  'suffix': PosixPath('html/home.html'),\n",
+       "  'type': 'html',\n",
+       "  'timestamp': 1707347952.187037}]"
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "root = Path(\"/app/work/crawl_data\")\n",
+    "files = find_files(root)\n",
+    "files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "id": "e57c0521-4eae-4323-b367-ea94a80938a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_os_client_args():\n",
+    "    args = {\n",
+    "        \"hosts\": [{\"host\": \"localhost\", \"port\": 9200}],\n",
```

Comment:
s/localhost/opensearch/
and add # Change this to localhost if you are running outside of the container.

Reply:
Alternately, and more complexly, check to see if you're in the container with a stat of /app/work/AAA_SEE_README_FOR_PERSISTENT_DATA_DIRECTORIES; if it exists it's in the container, otherwise outside.